public/js/index.js

```diff
@@ -0,0 +1,237 @@
+const keys = Object.keys(localStorage);
```

Comment:
Esse código funciona, porém ele tem alguns problemas, vamos lá:

1. Por você ter salvo cada produto em um local diferente do localStorage, você precisa ir várias vezes ao banco de dados para mostrar mais de um produto, o que normalmente não é bom, tá certo que é um banco de dados no navegador, mas isso consome mais recurso do que ir buscar a lista toda de uma vez, já que nossa lista não é gigantesca. 
2. Tem muito uso de querySelector e appendChild, que é uma operação relativamente cara, você poderia ter uma estratégia de criar o elemento todo de uma vez e fazer um querySelector e um appendChild no document uma vez só, no final de toda a categoria pronta. 
3. Se você realmente precisasse usar querySelector para encontrar um elemento na página, você poderia fazer também a partir de um elemento que você já tem a referencia, não precisa partir sempre do document, claro, se o elemento que você estiver procurando for filho do elemento que você já encontrou, isso reduz a escala da busca pelo navegador.